src/App.tsx

```diff
@@ -262,6 +272,50 @@ function AppContent() {
     }
   }, [type]);
 
+  useEffect(() => {
+    if (showUnitSnackbar) {
+      snackbar.pageInfo(
+        strings.formatString<any>(
+          strings.UNITS_INITIALIZED_MESSAGE,
+          userPreferences.preferredWeightSystem ?? 'metric',
+          <Link to={APP_PATHS.MY_ACCOUNT}>{strings.MY_ACCOUNT}</Link>
+        ),
+        strings.UNITS_INITIALIZED_TITLE,
+
+        {
+          label: strings.GOT_IT,
+          apply: () => {
+            updatePreferences('unitsAcknowledgedOnMs', Date.now());
+          },
+        },
+        'user'
+      );
+    }
+  }, [showUnitSnackbar, snackbar, userPreferences.preferredWeightSystem]);
+
+  useEffect(() => {
+    const initializaUnit = async () => {
+      if (!user || !userPreferences || !pathname) {
+        return;
+      }
+
+      if (!userPreferences.preferredWeightSystem) {
+        const response = await updatePreferences('preferredWeightSystem', 'metric');
+        if (response.requestSucceeded) {
+          setShowUnitSnackbar(true);
+        }
+      }
+
+      if (!userPreferences.unitsAcknowledgedOnMs) {
+        setShowUnitSnackbar(true);
+      }
+    };
```

Comment:
i have been moving logic like this into the api/user (or org depending on context) (which will be rewritten as a service soon).
fwiw - we have 2 ways of initializing .. time zone is done differently.. units done differently.. not sure if this can be done the same way for consistency.. if not no worries

Reply:
Ok! I tried to follow your implementation for consistency, but notification part is a bit different. Let me know what you think